### PR TITLE
feat: add created/dropped metrics to datablock

### PR DESCRIPTION
**Pull request by flaneur2020**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

this PR adds created/dropped metrics to DataBlock, we can get how many living DataBlocks by these two metrics, may help us diagnose some OOM issues.

### CODE REVIEW

File: common/datablocks/src/data_block.rs [hunk 1]
```diff
@@ -64,6 +66,7 @@ impl DataBlock {
             let col = f.data_type().create_column(&[]).unwrap();
             columns.push(col)
         }
+        super::metrics::incr_data_block_created();
```
Comment (BohuTANG) [inline common/datablocks/src/data_block.rs:69]: This metric `incr_data_block_created` add here is strange, can you explain why?

File: common/datablocks/src/data_block.rs [hunk 2]
```diff
@@ -254,8 +257,20 @@ impl fmt::Debug for DataBlock {
     }
 }
 
+impl Clone for DataBlock {
+    fn clone(&self) -> Self {
+        Self::create(self.schema.clone(), self.columns.clone())
```
Comment (BohuTANG) [inline common/datablocks/src/data_block.rs:262]: Do we need `super::metrics::incr_data_block_created();`?

Comment (flaneur2020) [reply common/datablocks/src/data_block.rs:262]: this clone() calls the `create` method, which contains a `incr_data_block_created()` inside 🤔

**Comment by everpcpc** [comment]
```

     Running tests/it/main.rs (target/debug/deps/it-d0e5e071410fc797)

running 50 tests
test api::http::cluster_state_test::test_cluster_nodes ... ok
test api::http::cluster_state_test::test_http_service_cluster_state ... ok
test api::http::config::test_config ... ok
test api::http::metrics::test_metrics ... ok
test api::http_service::test_http_service_tls_server ... ok
test configs::test_load_config ... ok
test configs::test_tls_rpc_enabled ... ok
test grpc::metasrv_grpc_api::test_auto_sync_addr ... ok
test grpc::metasrv_grpc_api::test_join ... ok
test grpc::metasrv_grpc_api::test_restart ... ok
test grpc::metasrv_grpc_api::test_retry_join ... ok
test grpc::metasrv_grpc_export::test_export ... ok
test grpc::metasrv_grpc_handshake::test_metasrv_handshake ... ok
test grpc::metasrv_grpc_kv_api::test_metasrv_kv_api ... ok
test grpc::metasrv_grpc_kv_api_restart_cluster::test_kv_api_restart_cluster_token_expired ... ok
test grpc::metasrv_grpc_kv_api_restart_cluster::test_kv_api_restart_cluster_write_read ... ok
test api::http::cluster_state_test::test_cluster_state has been running for over 60 seconds
test grpc::metasrv_grpc_schema_api::test_meta_grpc_client_cluster ... ok
test grpc::metasrv_grpc_schema_api::test_meta_grpc_client_single ... ok
test grpc::metasrv_grpc_schema_api_follower_follower::test_meta_grpc_client_database_create_get_drop ... ok
test grpc::metasrv_grpc_schema_api_follower_follower::test_meta_grpc_client_list_database ... ok
test grpc::metasrv_grpc_schema_api_follower_follower::test_meta_grpc_client_list_table ... ok
test grpc::metasrv_grpc_schema_api_follower_follower::test_meta_grpc_client_table_create_get_drop ... ok
test grpc::metasrv_grpc_schema_api_leader_follower::test_meta_api_database_create_get_drop ... ok
test grpc::metasrv_grpc_schema_api_leader_follower::test_meta_api_list_database ... ok
test grpc::metasrv_grpc_schema_api_leader_follower::test_meta_api_list_table ... ok
test grpc::metasrv_grpc_schema_api_leader_follower::test_meta_api_table_create_get_drop ... ok
test grpc::metasrv_grpc_tls::test_tls_client_config_failure ... ok
test grpc::metasrv_grpc_tls::test_tls_server ... ok
test grpc::metasrv_grpc_tls::test_tls_server_config_failure ... ok
test grpc::metasrv_grpc_watch::prefix_of_string_test ... ok
test grpc::metasrv_grpc_watch::test_get_start_and_end_of_prefix ... ok
test grpc::metasrv_grpc_watch::test_watch ... ok
test meta_node::meta_node_all::test_meta_node_boot ... ok
test meta_node::meta_node_all::test_meta_node_graceful_shutdown ... ok
test meta_node::meta_node_all::test_meta_node_incr_seq ... ok
test meta_node::meta_node_all::test_meta_node_join ... ok
test meta_node::meta_node_all::test_meta_node_join_rejoin ... ok
test meta_node::meta_node_all::test_meta_node_leader_and_non_voter ... ok
test meta_node::meta_node_all::test_meta_node_leave ... ok
test meta_node::meta_node_all::test_meta_node_restart ... ok
test meta_node::meta_node_all::test_meta_node_restart_single_node ... ok
test meta_node::meta_node_all::test_meta_node_snapshot_replication ... ok
test meta_node::meta_node_all::test_meta_node_write_to_local_leader ... ok
test meta_node::meta_node_kv_api::test_meta_node_kv_api ... ok
test store::test_impl_raft_storage ... ok
test store::test_meta_store_build_snapshot ... ok
test store::test_meta_store_current_snapshot ... ok
test store::test_meta_store_install_snapshot ... ok
test store::test_meta_store_restart ... ok
```


unit test stuck here for half an hour

**Comment by flaneur2020** [comment]
> DataBlock is a main path, I suggest not to add any metrics to the main branch, we can fork a special branch to debug it with metrics.

fair enough, however the metrics in databend is far not enough yet, I guess we can try lowering the cost of recording metrics & add more metrics to reflect the process of query execution.

currently the cost of updating a metric is equivant with updating an global atomic variable, we can try the approach of updating a thread local variable instead of an atomic one, 
 like describbes in https://archive.fosdem.org/2019/schedule/event/rust_prometheus/attachments/slides/3301/export/events/attachments/rust_prometheus/slides/3301/Introducing_Rust_Prometheus.pdf.

**Pull request closed without merging**